internal/gengapic/gengapic.go

```diff
@@ -62,11 +62,18 @@ func Gen(genReq *plugin.CodeGeneratorRequest) (*plugin.CodeGeneratorResponse, er
 
 	// parse plugin params, ignoring unknown values
 	for _, s := range strings.Split(*genReq.Parameter, ",") {
+		// check for the boolean flag, sample-only, that disables client generation
+		if s == "sample-only" {
+			return &g.resp, nil
+		}
+
 		e := strings.IndexByte(s, '=')
 		if e < 0 {
-			e = len(s)
+			return &g.resp, errors.E(nil, "invalid plugin option format, must be key=value: %s", s)
 		}
-		switch s[:e] {
+
+		key, val := s[:e], s[e+1:]
```

Comment:
Check `val == ""`, just in case? Won't panic, but won't be correct. Could also do `if e < 0 || e == len(s) - 1` above.

Reply:
I went with a `val == ""` check. Done.